articles/proofs-arguments-and-zero-knowledge/chapter_12.md

```diff
@@ -0,0 +1,289 @@
+$z$ in $\lbrace 0,...,|G| −1 \rbrace$ and the commitment is $g^m \cdot h^z$ . One thinks of $h^z$ as a random group element that
+operates as a
+“blinding factor”: by multiplying $g^m$ by $h^z$ , one ensures that the commitment is a random group element, statistically independent
+of $m$. Below is the traditional presentation of Pedersen commitments:
+
+---
+
+**Protocol 5** Standard presentation of Pedersen commitments in a cyclic group $G$ for which the Discrete Logarithm problem is
+intractable.
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order. The key generation procedure publishes randomly chosen generators $g,h
+\in G$, which serve as both the commitment key and verification key.  
+$2$: To commit to a number $m \in \lbrace 0,...,|G| −1 \rbrace$, committer picks a random $z \in \lbrace 0,...,|G| −1 \rbrace$ and
+sends $c \leftarrow{} g^m
+\cdot h^z$ .  
+$3$: To open a commitment $c$, committer sends $(m,z)$. Verifier checks that $c = g^m \cdot h^z$ .
+
+---
+
+Pedersen commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation:
+
+---
+
+**Protocol 6** Commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation.
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order.  
+$2$: The key generation procedure publishes randomly chosen generators $g,h \in G$, which serve as both the commitment key and
+verification key.  
+$3$: To commit to a number $m \in \lbrace 0,...,|G| −1 \rbrace$, committer picks a random $z \in \lbrace 0,...,|G| −1 \rbrace$ and
+sends $c \leftarrow{}
+h^{−m} \cdot g^z$  
+$4$: To open a commitment $c$, committer sends $(m,z).$ Verifier checks that $c \cdot h^m = g^z$
+
+---
+
+### 12.3.1 Important Properties of Pedersen Commitments
+
+**Additive Homomorphism**: This means that the verifier can take two commitments $c_1$ and $c_2$, to values $m_1,m_2 \in \lbrace
+0,...,|G| −
+1 \rbrace$ (with $m_1,m_2$ known to the committer but not to the verifier), and the verifier on its own can derive a commitment $c_3$
+to $m_3
+:= m_1 + m_2$, such that the prover is able to open $c_3$ to $m_3$. This is done by simply letting $c_3 \leftarrow{} c_1 \cdot
+c_2$. As for "opening information" provided by the prover, if $c_1 = h^{m_1} \cdot g^{z_1}$ and $c_2 = h^{m_2} \cdot g^{z_2}$ ,
+then $c_3 = h^{m_1+m_2} \cdot g^{z_1+z_2}$, so the opening information for $c_3$ is simply $(m_1 + m_2,z_1 + z_2).$
+
+**Perfect HVZK Proof of Knowledge of Opening**: It will be useful for the prover to prove that it knows how to open a commitment $c$ to
+some value, without actually opening the commitment. Let's see it:
+
+---
+
+**Protocol 7** Zero-Knowledge Proof of Knowledge of Opening of Pedersen Commitment
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order over which the Discrete Logarithm relation is hard, with randomly chosen
```

Comment:
Place algorithm in something like this:

```
abd
```

Reply:
Place these algorithms in block will raise math formula errors, I don't know a better way to deal with it.